crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -272,7 +283,10 @@ async fn download_block(
                 "Block hash mismatch"
             );
             match block.status {
-                Status::AcceptedOnL1 | Status::AcceptedOnL2 => Ok(DownloadBlock::Block(block)),
+                Status::AcceptedOnL1 | Status::AcceptedOnL2 => Ok(DownloadBlock::Block(
+                    block,
+                    verify_result.get_commitments().unwrap_or_default(),
```

Comment:
Good point! Agree, `get_commitments()` is unnecessary, matching the `verify_result` in-place seems more readable.

Seems like `anyhow::ensure!` is just a "sweep under the rug" approach to distinguish different modes for downloading blocks (strict and non-strict ones). Maybe I got carried away a bit, but this is how I think it can be solved: https://github.com/eqlabs/pathfinder/pull/835.

Reply:
I see why you think #835 makes sense, but in this PR I think maybe simply keeping the same behavior without using `anyhow::ensure!()` would be sufficient?